internal/provider/resource_jaas_group_test.go

```diff
@@ -0,0 +1,173 @@
+			matches: true,
+		},
+		{
+			desc:  "Invalid - starts with a number",
+			input: "1foo",
+		},
+		{
+			desc:  "Invalid - starts with a hyphen",
+			input: "-foo",
+		},
+		{
+			desc:  "Invalid - ends with a hyphen",
+			input: "foo-",
+		},
+		{
+			desc:  "Invalid - ends with special character",
+			input: "foo@",
+		},
+		{
+			desc:  "Invalid - contains special character",
+			input: "foo@bar",
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			assert.Equal(t, groupNameRe.MatchString(tC.input), tC.matches)
+		})
+	}
+}
+
+func TestAcc_ResourceJaasGroup(t *testing.T) {
+	OnlyTestAgainstJAAS(t)
+	groupName := acctest.RandomWithPrefix("tf-jaas-group")
+	newGroupName := acctest.RandomWithPrefix("tf-jaas-group-new")
+	resourceName := "juju_jaas_group.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		CheckDestroy:             testAccCheckJaasGroupDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceJaasGroup(groupName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", groupName),
+					resource.TestCheckResourceAttrSet(resourceName, "uuid"),
+					testAccCheckJaasGroupExists(resourceName),
+				),
+			},
+			{
+				Config: testAccResourceJaasGroup(newGroupName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", newGroupName),
+					resource.TestCheckResourceAttrSet(resourceName, "uuid"),
+					testAccCheckJaasGroupExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_ResourceJaasGroupInvalid(t *testing.T) {
+	OnlyTestAgainstJAAS(t)
+	invalidGroupName := "_invalid name"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceJaasGroup(invalidGroupName),
+				// Might break if the formatting changes
+				ExpectError: regexp.MustCompile("must start with a letter, end with a letter or number"),
+			},
+		},
+	})
+}
+
+func testAccResourceJaasGroup(name string) string {
+	return fmt.Sprintf(`
+resource "juju_jaas_group" "test" {
+  name = %q
+}`, name)
+}
+
+func testAccCheckJaasGroupExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Group %q not found", resourceName)
+		}
+
+		uuid := rs.Primary.Attributes["uuid"]
+		if uuid == "" {
+			return errors.New("No group uuid is set")
+		}
+
+		_, err := TestClient.Jaas.ReadGroup(context.Background(), uuid)
+		if err != nil {
+			return fmt.Errorf("Error fetching group with uuid %q: %s", uuid, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckJaasGroupDestroyed(s *terraform.State) error {
```

Comment:
Maybe rename the function to `testAccCheckAllJaasGroupsDestroyed` since we're checking if ALL groups were destroyed.

Reply:
Thinking some more, I think this is the wrong thing to do, because if we ran tests in parallel this would break. Rather, just check if the resource you created in your test no longer exists.